nodejs/lesson03/solution/formatter.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const express = require('express')
+const app = express()
+const initTracer = require('../../lib/tracing').initTracer;
+const { Tags, FORMAT_HTTP_HEADERS } = require('opentracing');
+
+const tracer = initTracer('format-service');
+
+const port = 8081;
+
+app.listen(port, function () {
+    console.log('Formatter app listening on port ' + port);
+})
+
+app.get('/format', function (req, res) {
+    const parentSpanContext = tracer.extract(FORMAT_HTTP_HEADERS, req.headers)
+    const span = tracer.startSpan('http_server', {
+        childOf: parentSpanContext
+    });
+    span.setTag(Tags.SPAN_KIND, Tags.SPAN_KIND_RPC_SERVER);
```

Comment:
there is a certain quirk when people use Zipkin, so the preferred way is to pass this particular tag into startSpan call, as `tags: {Tags.SPAN_KIND: Tags.SPAN_KIND_RPC_SERVER}` field in the options argument (next to childOf)